### DISCUSSION TITLE
Custom / non default fonts not showing up ?

### DISCUSSION THREAD

**Discussion by rdekens** [moononournation/Arduino_GFX#400] [General]
Using a canvas as a "sprite" within the display - 200x200 display that is emulating a EPD display from another project.
the backgound is white - and writing black text.
canvas-> setFont(......)
canvas -> setTextColor(BLACK, WHITE);
canvas->print( .....) 

If i print after a setFont();  using the default font - I get small black text as expected - 
but If i use one of the standard included fonts eg FreeMonoBold9pt7b.h
then no text is seen  the only text that shows up is default font.

anyone seen this 
I've got to dig a bit deeper yet - but it looks like something in the inheritance of the print function ?
Ray D

**Comment by moononournation** [comment]
try set cursor to a lower position.

**Comment by rdekens** [comment]
ok will give it a go - but coordinates are the same ( same code) from the EPD .... 
I would have thought the text should do an overwrite at position anyway - not some strange collision detection.
I will recheck order of drawing, but the text should be last .. so I would expect it to overwrite - 

will check tonight

**Comment by rdekens** [comment]
moving the text location has no effect - works with the default font using write and print - doesnt work with any oft the other fonts.
As near as i can gather its an issue to do with the definition of the fonts / or the virtualisation of them in the inheritance - it seems to be an issue with the arduino gfx library structure.
It seems to be a function of the fact that I'm using a class to implement the "display"  which then uses a canvas class . In this case the fonts need to be referenced within the "display" class and this seems to be the problem.

Im a bit out of my depth at the moment.
ray

**Comment by moononournation** [comment]
Just tested with HelloWorldGfxfont.ino example, everything works fine:
```
Arduino_G *gfx = new Arduino_ILI9342(bus, 48 /* RST */, 0 /* rotation */);
Arduino_GFX *canvas = new Arduino_Canvas(320 /* width */, 240 /* height */, gfx);
```
![IMG_7299](https://github.com/moononournation/Arduino_GFX/assets/13228891/00f97e4f-7210-4403-8a89-aedc03d91837)

**Reply by moononournation** [reply]
remember calling canvas->flush() after draw finished.

**Comment by rdekens** [comment]
I'll modify an example tonight using the hello world gif - but with the appropriate class and canvas setup to test and show the problem and post it back for you to test. 

I'm half way there but had to go to the day job. ...

Ray

**Comment by rdekens** [comment]
[HelloWorldGfxfont.zip](https://github.com/moononournation/Arduino_GFX/files/13670987/HelloWorldGfxfont.zip)

this should illustrate - used on a lilygo-T-Display-S3 but should be no different .
compare with SUB_WINDOW defined and not defined and you should see the issue

Ray

**Comment by rdekens** [comment]
interestingly - doesn't just happen when encapsulated in a class - its more serious than that - Ive flattened it outside of a class - 
drawing objects works but anything other than the default font doesn't work when a canvas is set to output to another canvas - 

This makes it more significant
flat version attached
[HelloWorldGfxfont_flat.zip](https://github.com/moononournation/Arduino_GFX/files/13679142/HelloWorldGfxfont_flat.zip)
